src/Elm.elm

```diff
@@ -1156,6 +1156,135 @@ lambda argBaseName argType toExpression =
+                    Err err
+
+                Ok return ->
+                    List.foldr
+                        (\ann fnbody ->
+                            Annotation.FunctionTypeAnnotation
+                                (Compiler.nodify ann)
+                                (Compiler.nodify fnbody)
+                        )
+                        return
+                        [ Compiler.getInnerAnnotation argType ]
+                        |> Ok
+        , imports = expr.imports
+        , skip = False
+        }
+
+
+betaReduce : Exp.Expression -> Exp.Expression
+betaReduce e =
+    let
+        extractLastArg arg =
+            case arg of
+                Exp.FunctionOrValue [] n ->
+                    Just n
+
+                Exp.ParenthesizedExpression p ->
+                    extractLastArg <| Compiler.denode p
+
+                _ ->
+                    Nothing
+
+        -- If the list is nonempty, returns a tuple with the beginning of the list and the last element (denoded).
+        popLast : List (Node.Node a) -> Maybe ( List (Node.Node a), a )
+        popLast lst =
+            case List.reverse lst of
+                [] ->
+                    Nothing
+
+                last :: initReverse ->
+                    Just ( List.reverse initReverse, Compiler.denode last )
+    in
+    case e of
+        Exp.LambdaExpression { args, expression } ->
+            case popLast args of
+                Just ( initLambdaArgs, Pattern.VarPattern lastLambdaArg ) ->
+                    case Compiler.denode expression of
+                        Exp.RecordAccess argNode fieldNode ->
+                            let
+                                fieldName =
+                                    Compiler.denode fieldNode
+
+                                arg =
+                                    Compiler.denode argNode
+                            in
+                            case arg of
+                                Exp.FunctionOrValue [] argName ->
+                                    if argName == lastLambdaArg then
+                                        Exp.RecordAccessFunction <| "." ++ fieldName
+
+                                    else
+                                        e
+
+                                _ ->
+                                    e
+
+                        Exp.Application applicationArgs ->
+                            case popLast applicationArgs of
+                                Just ( [], uniqueApplicationArg ) ->
+                                    if extractLastArg uniqueApplicationArg == Just lastLambdaArg then
+                                        Exp.FunctionOrValue [] "identity"
+
+                                    else
+                                        e
+
+                                Just ( initApplicationArgs, lastApplicationArg ) ->
+                                    if extractLastArg lastApplicationArg == Just lastLambdaArg then
```

Comment:
This `if` is the same as the `if` above it, which makes my ~spider~ smell sense tingle